Training/Repository/IRepository.cs

```diff
@@ -4,7 +4,8 @@ public interface IRepository<T> where T: class {
     Task<T> Add(T entity);
     Task<T> Update(T entity);
     Task Delete(T entity);
-    Task<T?> GetById(Guid id);
+    Task<T> GetById(Guid id);
     Task<IEnumerable<T>> GetAll();
+    Task<T> DeleteWithoutDeleteing(Guid id);
```

Comment:
<T> is not needed on delete Task is enough

Reply:
you said that you want a delete that actually deletes the cows from the database

Reply:
refactor with Task instead of Task<T>